src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -2167,6 +2167,212 @@ protected override void BeginProcessing()
+
+        #endregion Parameters
+
+        #region Overrides
+        /// <summary>
+        /// Create the service
+        /// </summary>
+        [ArchitectureSensitive]
+        protected override void ProcessRecord()
+        {
+            ServiceController service = null;
+            string ServiceComputerName = null;
+            foreach (string computer in ComputerName)
+            {
+                bool objServiceShouldBeDisposed = false;
+                try
+                {
+                    if (_ParameterSetName.Equals("InputObject", StringComparison.OrdinalIgnoreCase) && InputObject != null)
+                    {
+                        service = InputObject;
+                        Name = service.ServiceName;
+                        ServiceComputerName = service.MachineName;
+                        objServiceShouldBeDisposed = false;
+                    }
+                    else
+                    {
+                        ServiceComputerName = computer;
+                        service = new ServiceController(Name, ServiceComputerName);
+                        objServiceShouldBeDisposed = true;
+                    }
+                    Diagnostics.Assert(!String.IsNullOrEmpty(Name), "null ServiceName");
+                    // "new ServiceController" will succeed even if
+                    // there is no such service.  This checks whether
+                    // the service actually exists.
+
+                    string unusedByDesign = service.DisplayName;
+                }
+                catch (ArgumentException ex)
+                {
+                    //Cannot use WriteNonterminatingError as service is null
+                    ErrorRecord er = new ErrorRecord(ex, "ArgumentException", ErrorCategory.ObjectNotFound, computer);
+                    WriteError(er);
+                    continue;
+                }
+                catch (InvalidOperationException ex)
+                {
+                    //Cannot use WriteNonterminatingError as service is null
+                    ErrorRecord er = new ErrorRecord(ex, "InvalidOperationException", ErrorCategory.ObjectNotFound, computer);
+                    WriteError(er);
+                    continue;
+                }
+
+                try // In finally we ensure dispose, if object not pipelined.
+                {
+                    // Confirm the operation first.
+                    // This is always false if WhatIf is set.
+                    if (!ShouldProcessServiceOperation(service))
+                    {
+                        continue;
+                    }
+
+                    NakedWin32Handle hScManager = IntPtr.Zero;
+                    NakedWin32Handle hService = IntPtr.Zero;
+                    try
+                    {
+                        hScManager = NativeMethods.OpenSCManagerW(
+                            ServiceComputerName,
+                            null,
```

Comment:
Please use named parameters.

Reply:
I don't quite follow you. Did you want to use the variable `ComputerName` ? Since the method supports both a `Name` on the local computer and a servicecontroller on a remote machine, I believe I have to use the `ServiceComputerName` variable.

Reply:
see https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments

It makes the code more readable, particularly where null is passed.